Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/DictionaryWorkspaceController.java

```diff
@@ -31,265 +30,200 @@
-	 */
-	@AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE,paramIndex = 1, userRole = {RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN } )})
-	@RequestMapping(value = "auth/workbench/workspace/{workspaceid}/adddictionary", method = RequestMethod.GET)
-	public String addWorkspaceDictionary(
-			@PathVariable("workspaceid") String workspaceId, Model model) 
-					throws QuadrigaStorageException, QuadrigaAccessException
-	{
-		try {
-			UserDetails user = (UserDetails) SecurityContextHolder.getContext()
-					.getAuthentication().getPrincipal();
-			String userId = user.getUsername();
-			logger.info("USer : " + user.getUsername()
-					+ " trying to add dictionary into workspace");
-			List<IDictionary> dictionaryList = null;
-			try {
-				dictionaryList = workspaceDictionaryManager.getNonAssociatedWorkspaceDictionaries(
-						workspaceId, userId);
-			} catch (QuadrigaStorageException e) {
-				throw new QuadrigaStorageException(
-						"Oops the DB is an hard hangover, please try later");
-			}
-			if (dictionaryList == null) {
-				logger.info("Dictionary list is empty");
-			}
-			model.addAttribute("dictinarylist", dictionaryList);
-			model.addAttribute("workspaceId", workspaceId);
-			IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId,userId);
-			model.addAttribute("workspacedetails", workspace);
-			model.addAttribute("userId", userId);
-		} catch (Exception e) {
-			logger.error("issue while adding dictionary " ,e);
-		}
-		return "auth/workbench/workspace/adddictionaries";
-	}
+    /**
+     * Retrieve dictionaries from the system to associate the selected
+     * dictionaries with the given workspace
+     * 
+     * @param workspaceId
+     * @param model
+     * @return String - URL to redirect on success or failure
+     * @throws QuadrigaStorageException
+     * @throws QuadrigaAccessException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.WORKSPACE, paramIndex = 1, userRole = {
+            RoleNames.ROLE_WORKSPACE_COLLABORATOR_ADMIN }) })
+    @RequestMapping(value = "auth/workbench/workspace/{workspaceid}/adddictionary", method = RequestMethod.GET)
+    public String addWorkspaceDictionary(@PathVariable("workspaceid") String workspaceId, Model model,
+            Principal principal) throws QuadrigaStorageException, QuadrigaAccessException {
+        String userId = principal.getName();
+        List<IDictionary> dictionaryList = null;
+        dictionaryList = workspaceDictionaryManager.getNonAssociatedWorkspaceDictionaries(workspaceId, userId);
+        if (dictionaryList == null) {
+            logger.info("Dictionary list is empty");
+        }
+        model.addAttribute("dictinarylist", dictionaryList);
+        model.addAttribute("workspaceId", workspaceId);
+        IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId, userId);
+        model.addAttribute("workspacedetails", workspace);
+        model.addAttribute("userId", userId);

```

Comment:
is this actually used in the jsp? This should be available through spring security if needed.